src/agent/coverage/src/code.rs

```diff
@@ -302,83 +301,113 @@ impl Region for Symbol {
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(transparent)]
-pub struct SymbolFilterSpec {
-    filters: HashMap<String, Filter>,
+pub struct CmdFilterDef {
+    defs: Vec<ModuleRuleDef>,
 }
 
-#[derive(Clone, Debug)]
-pub struct SymbolFilter {
-    /// Pre-compiled regex set for fast matching.
-    regexes: RegexSet,
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct ModuleRuleDef {
+    pub module: String,
 
-    /// Maps module names (via regexes) to filters for the symbol names provided
-    /// by the matched module(s).
+    #[serde(flatten)]
+    pub rule: RuleDef,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
```

Comment:
Add comment explaining how this type exists to be a serde target and support a slightly more complex (and intuitive) user-facing file format.